mapper/tedge_mapper/src/sm_c8y_mapper/tests.rs

```diff
@@ -292,31 +224,140 @@ async fn mapper_publishes_software_update_status_and_software_list_onto_c8y_topi
             ],
             "failures":[]
         }"#;
+
     let _ = publish(
         &Topic::new("tedge/commands/res/software/update").unwrap(),
         json_response.to_string(),
     )
     .await;
 
-    // Expect `116` and `502` messages with correct payload have been received on `c8y/s/us`, if no msg received for the timeout the test fails.
+    // `502` messages with correct payload have been received on `c8y/s/us`, if no msg received for the timeout the test fails.
     let mut received_status_failed = false;
-    let mut received_software_list = false;
-    for _ in 0..2 {
+
+    for _ in 0..4 {
         match tokio::time::timeout(TEST_TIMEOUT_MS, subscriber.next()).await {
             Ok(Some(msg)) => {
                 dbg!(&msg.payload_str().unwrap());
                 match msg.payload_str().unwrap() {
-                        "502,c8y_SoftwareUpdate,\"Partial failure: Couldn\'t install collectd and nginx\"\n" => received_status_failed = true,
-                        "116,nginx,1.21.0::docker,\n" => received_software_list = true,
+                        "502,c8y_SoftwareUpdate,\"Partial failure: Couldn\'t install collectd and nginx\"\n" => { received_status_failed = true; break;}
                         _ => {}
                     }
                 continue;
             }
-            _ => panic!("No message received after a second."),
+            _ => panic!("No failed status message received after a second."),
         }
     }
     assert!(received_status_failed);
-    assert!(received_software_list);
+}
+
+#[tokio::test]
+#[cfg_attr(not(feature = "mosquitto-available"), ignore)]
+#[serial]
+async fn mapper_fails_during_sw_update_recovers_and_process_response() -> Result<(), anyhow::Error>
+{
+    // The test assures SM Mapper correctly receives software update request smartrest message on `c8y/s/ds`
+    // and converts it to thin-edge json message published on `tedge/commands/req/software/update`.
+    // SM Mapper fails before receiving the rsponse for the request.
+    // Meanwhile the opeartion response was published on `tedge/commands/res/software/update`.
+    // Now the SM Mapper recovers and receives the response message and publishes it on `c8y/s/us`
+    // The subscriber that was waiting for the response on `c8/s/us` receives the response and validates it.
+
+    // Create a subscriber to receive messages on `tedge/commands/req/software/update` topic.
+    let mut sw_update_req_sub = get_subscriber(
+        "tedge/commands/req/software/update",
+        "mapper_publishes_software_update_request",
+    )
+    .await;
+
+    // Create a subscriber to receive messages on `"c8y/s/us` topic.
+    let mut sw_update_res_sub = get_subscriber(
+        "c8y/s/us",
+        "mapper_publishes_software_update_response_to_c8y_cloud",
+    )
+    .await;
+
+    // Start SM Mapper
+    let sm_mapper = start_sm_mapper().await?;
+
+    // Prepare and publish a software update response `successful`.
```

Comment:
The comment is misleading: the response is not published - at least not now.

Since a key point of that test is to ensure the response will be read even if sent when the mapper is down, this needs to be clearer.